go/parquet/file/column_writer_types.gen.go.tmpl

```diff
@@ -193,14 +193,23 @@ func (w *{{.Name}}ColumnChunkWriter) WriteDictIndices(indices arrow.Array, defLe
 
     valueOffset += info.numSpaced()
   })
-  
+
   return
 }
 
 func (w *{{.Name}}ColumnChunkWriter) writeValues(values []{{.name}}, numNulls int64) {
   w.currentEncoder.(encoding.{{.Name}}Encoder).Put(values)
   if w.pageStatistics != nil {
+{{- if ne .Name "FixedLenByteArray"}}
     w.pageStatistics.(*metadata.{{.Name}}Statistics).Update(values, numNulls)
+{{- else}}
+    s, ok := w.pageStatistics.(*metadata.{{.Name}}Statistics)
+    if ok {
+      s.Update(values, numNulls)
+    } else {
+      w.pageStatistics.(*metadata.Float16Statistics).Update(values, numNulls)
+    }
+{{- end}}
```

Comment:
would probably be better to check `w.Descr().LogicalType()` and check if it's a Float16 column rather than make this assumption which might bite us in the future.

Reply:
Updated now, but worth another look in case I misunderstood the concern.

Reply:
looks good! thanks!